Assets/BowAndArrowImproved/Scripts/TeleportAnchor.cs

```diff
@@ -0,0 +1,30 @@
+﻿using UnityEngine;
+using UnityEngine.Events;
+
+public class TeleportAnchor : MonoBehaviour
```

Comment:
this script is not a "TeleportAnchor", this script has absolutely nothing to do with teleportation, it simply fires an event on collision, if all you want to know is if an arrow collided with a "teleport anchor", you can simply put the "anchors" as child objects of the teleport manager and add an `OnCollisionEnter` function to the teleport manager - unity tells the parent object about any collisions happening in it's children, read more about it